src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/migration/Cas1UserDetailsMigrationJob.kt

```diff
@@ -0,0 +1,147 @@
+      createApplicantUserDetailsEntity(application, confirmYourDetailsMap, application.applicantUserDetails?.id),
+    )
+
+    val applicantIsCaseManager = confirmYourDetailsMap == null ||
+      !confirmYourDetailsMap.containsKey("caseManagementResponsibility") ||
+      confirmYourDetailsMap["caseManagementResponsibility"] == "yes"
+
+    application.caseManagerIsNotApplicant = !applicantIsCaseManager
+
+    if (!applicantIsCaseManager) {
+      application.caseManagerUserDetails = cas1UserDetailsRepository.save(
+        createCaseManagerUserDetails(basicInformationMap, application, application.caseManagerUserDetails?.id),
+      )
+    } else {
+      application.caseManagerUserDetails = null
+    }
+
+    applicationRepository.save(application)
+
+    log.info("Contact details updated for application ${application.id}. Applicant is case manager? $applicantIsCaseManager")
+  }
+
+  private fun createCaseManagerUserDetails(
+    basicInformationMap: Map<*, *>?,
+    application: ApprovedPremisesApplicationEntity,
+    existingId: UUID?,
+  ): Cas1ApplicationUserDetailsEntity {
+    val caseManagerDetails = basicInformationMap?.get("case-manager-information") as Map<*, *>?
+
+    val name = caseManagerDetails?.get("name") as String?
+      ?: error("caseManagementResponsibility is 'no', but no value defined for 'case-manager-information/name' for application ${application.id}")
+
+    val emailAddress = caseManagerDetails?.get("emailAddress") as String?
+      ?: error("caseManagementResponsibility is 'no', but no value defined for 'case-manager-information/emailAddress' for application ${application.id}")
+
+    val telephoneNumber = caseManagerDetails?.get("phoneNumber") as String?
+      ?: error("caseManagementResponsibility is 'no', but no value defined for 'case-manager-information/phoneNumber' for application ${application.id}")
+
+    return Cas1ApplicationUserDetailsEntity(
+      existingId ?: UUID.randomUUID(),
+      name = name,
+      email = emailAddress,
+      telephoneNumber = telephoneNumber,
+    )
+  }
+
+  private fun createApplicantUserDetailsEntity(
+    application: ApprovedPremisesApplicationEntity,
+    confirmYourDetailsMap: Map<*, *>?,
+    existingId: UUID?,
+  ): Cas1ApplicationUserDetailsEntity {
+    val applicantUserEntity = application.createdByUser
+    val detailsUptoDateMap = confirmYourDetailsMap?.get("detailsToUpdate") as List<*>?
+
+    val applicantNameOverridden = detailsUptoDateMap?.contains("name") ?: false
+    val applicantEmailOverridden = detailsUptoDateMap?.contains("emailAddress") ?: false
+    val applicantPhoneOverridden = detailsUptoDateMap?.contains("phoneNumber") ?: false
+
+    val applicantName = if (applicantNameOverridden) {
+      if (confirmYourDetailsMap!!.contains("name")) {
+        confirmYourDetailsMap["name"]!! as String
+      } else {
+        error("applicant name marked as overridden, but no value defined for 'confirm-your-details/name' for application ${application.id}")
```

Comment:
this looks robust, it will be interesting to see if you've foreseen all the possible issues!